src/Chunk.cpp

```diff
@@ -139,7 +139,8 @@ cChunk::~cChunk()
 	{
 		if (!(*itr)->IsPlayer())
 		{
-			(*itr)->Destroy(false);
+			// Scheduling a normal destruction is neither possible (Since this chunk will be gone till the schedule occurs) nor necessary.
+			(*itr)->DestroyNoScheduling(false);  // No point in broadcasting in an unloading chunk. Chunks unload when no one is nearby.
```

Comment:
Ah, whilst I concur that pointers are indeed faster, I am of the humble opinion that their complex manipulation is plainly unsafe, speed notwithstanding ;)



On 13 Feb 2016, at 18:48, "LogicParrot" <notifications@github.com> wrote:

> In src/Chunk.cpp:
> 
> > @@ -139,7 +139,8 @@ cChunk::~cChunk()
> >  	{
> >  		if (!(*itr)->IsPlayer())
> >  		{
> > -			(*itr)->Destroy(false);
> > +			// Scheduling a normal destruction is neither possible (Since this chunk will be gone till the schedule occurs) nor necessary.
> > +			(*itr)->DestroyNoScheduling(false);  // No point in broadcasting in an unloading chunk. Chunks unload when no one is nearby.
> But DoWithChunk is both slower and not future-safe!
> 
> —
> Reply to this email directly or view it on GitHub.
>